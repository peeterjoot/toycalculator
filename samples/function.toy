FUNCTION plus3( INT32 v ) : INT32
{
    INT32 r;
    r = 3 + v;
    RETURN r;
};

FUNCTION zero( ) : INT32
{
    RETURN 0;
};

FUNCTION add( INT32 v, INT32 w ) : INT32
{
    INT32 r;
    r = v + w;
    RETURN r;
};

FUNCTION bar0 ( )
{
    PRINT "bar0";
    RETURN;
};

FUNCTION bar1 ( )
{
    PRINT "bar1";
    RETURN;
};

PRINT "Hello functions.";

CALL bar0();
CALL bar1();

INT32 v;
v = CALL add( 42, 0 );
PRINT v;
v = CALL zero();
PRINT v;
v = CALL plus3( 39 );
PRINT v;

PRINT "Goodbye functions.";
