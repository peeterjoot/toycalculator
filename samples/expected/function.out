Hello functions.
bar0
bar1
42
0
42
Goodbye functions.
